app/javascript/components/TenantProfilePage.js

```diff
@@ -5,22 +5,110 @@ class TenantProfilePage extends Component {
     super(props);
     this.state = {
       tenant_id: this.props.id,
-      tenant_info: {"name": "", "email": "", "landowner_id": ""}
+      name: "", 
+      email: "", 
+      landowner_id: "",
+      street_address: "",
+      city: "",
+      zip: "",
+      state: "",
+      changed: false
     }
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
   }
 
   componentDidMount() {
     fetch("/tenants/" + this.state.tenant_id)
       .then((response) => {return response.json()})
-      .then((data) => {this.setState({ tenant_info: data})});
+      .then((data) => {this.setState({
+        name: data.name, 
+        email: data.email, 
+        landowner_id: data.landowner_id,
+        street_address: data.street_address,
+        city: data.city,
+        zip: data.zip,
+        state: data.state
+      })});
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    // Todo: change the values in the database

```

Comment:
Is this going to make some kind of POST/PATCH request? If so, are we going to make that API in this PR or for another?

Reply:
Yes it will, I'm planning to make that in another pr.